microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,314 @@
+
+The data has two main parts:
+
+* The metadata, which contains all of the information you need in order to parse the arrays of data found in the payload to re-create the graph.
+* The payload, which contains the actual data needed to re-create the graph.
+
+Note that the format of the `heapsnapshot` file, as documented below, may change in the future as V8 and DevTools evolve. We always try to keep our documentation up to date but if you encounter a discrepancy please send us feedback on our [GitHub repository](https://github.com/MicrosoftDocs/edge-developer/issues).
+
+
+<!-- ====================================================================== -->
+## Schema of the `heapsnashot` data
+
+### Top-level structure
+
+The `heapsnapshot` JSON data contains a root object with the following properties:
+
+```json
+{
+    "snapshot": {},
+    "nodes": [],
+    "edges": [],
+    "trace_function_infos": [],
+    "trace_tree": [],
+    "samples": [],
+    "locations": [],
+    "strings": []
+}
+```
+
+| Property | Description | Format |
+| --- | --- | --- |
+| `snapshot` | Contains all the information about the format of the payload and its size | `Object` |
+| `nodes` | All the information required to re-create the nodes of the graph. To parse this information you need `snapshot.meta.node_types` and `snapshot.meta.node_fields` | `Array` |
+| `edges` | All the information required to re-create the edges of the graph. To parse this you need `snapshot.meta.edge_types` and `snapshot.meta.edge_fields` | `Array` |
+| `trace_function_infos` | _Not documented yet_ | `Array` |
+| `trace_tree` | _Not documented yet_ | `Array` |
+| `samples` | _Not documented yet_ | `Array` |
+| `locations` | _Not documented yet_ | `Array` |
+| `strings` | An array with all the strings held in memory. These can be user-defined strings, code, etc. | `Array` |
+
+### Snapshot
+
+```json
+{
+    "snapshot": {     
+        "meta": {},
+        "node_count": 123,
+        "edge_count": 456,
+        "trace_function_count": 0
+    }
+    ...
+}
+```
+
+| Property | Description | Format |
+| --- | --- | --- |
+| `meta` | Contains several fields with information about the shape and size of every object contained in the payload | `Object` |
+| `node_count` | The total number of nodes | `Number` |
+| `edge_count` | The total number of edges. | `Number` |
+| `trace_function_count` | The total number of trace functions | `Number` |
+
+### Snapshot metadata
+
+```json
+{
+    "snapshot": {
+        "meta": {
+            "node_fields": [],
+            "node_types": [],
+            "edge_fields": [],
+            "edge_types": []
+        }
+    }
+    ...
+}
+```
+
+| Property | Description | Format |
+| --- | --- | --- |
+| `node_fields` | The list of all the properties required to recreate a node. | `Array` |
```

Comment:
Global: Consistently write "re-create" or "recreate".  I'd write "re-create"; clearer.  One meaning of "recreate" is recreation.  re-create is unambiguous.

https://www.ahdictionary.com/word/search.html?q=recreate